homework-testing/tests/binarySearchTree.test.js

```diff
@@ -0,0 +1,254 @@
+const BinarySearchTree = require('../src/binary-search-tree');
+const valuesArray = require('./values');
+
+describe('Binary Search Tree', () => {
+  let bst;
+  beforeEach(() => {
+    bst = new BinarySearchTree();
+  });
+
+  describe('#roots', () => {
+    it('Roots Method should return root of the tree', () => {
+      const key = 10;
+      const value = '10';
+      bst.insert(key, value);
+      const result = bst.roots();
+      expect(result).toEqual(value);
+    });
+
+    it('Roots Method should return null if tree is empty', () => {
+      const result = bst.roots();
+      expect(result).toEqual(null);
+    });
+  });
+
+  describe('#insert', () => {
+    it('Insert Method should fill the root if tree is empty', () => {
+      const key = 10;
+      const value = '10';
+      bst.insert(key, value);
+      const result = bst.roots();
+      expect(result).toEqual(value);
+    });
+
+    it('Insert Method should fills deep', () => {
+      valuesArray.forEach((element) => {
+        bst.insert(element.key, element.value);
+      });
+      const sortedArray = bst.traverse(true);
```

Comment:
your test should be as much independent as possible, what if `traverse` works incorrectly?